### PR TITLE
Added Data Structures and Algorithms to the cheatsheet

### PR DESCRIPTION
**Pull request by gupta-soham**
## What does this PR do?
Add resource(s) | Remove resource(s) | Add info | Improve repo

## Checklist:
- [x] Read our [contributing guidelines](https://github.com/EbookFoundation/free-programming-books/blob/main/docs/CONTRIBUTING.md).
- [x] [Search](https://ebookfoundation.github.io/free-programming-books-search/) for duplicates.
- [x] Include author(s) and platform where appropriate.
- [x] Put lists in alphabetical order, correct spacing.
- [x] Add needed indications (PDF, access notes, under construction).
- [x] Used an informative name for this pull request.

## Follow-up

- Check the status of GitHub Actions and resolve any reported warnings!

### CODE REVIEW

**Comment by eshellman** [comment]
Be sure to check the free-programming-books-lint output for linter errors. Alphabetical order and spacing are the most common problems.

**Comment by gupta-soham** [comment]
How do I fix this error @eshellman

File: more/free-programming-cheatsheets.md [hunk 1]
```diff
@@ -81,6 +82,15 @@
 * [Cheatsheets for Data Scientists](https://www.datacamp.com/community/data-science-cheatsheets) - Datacamp (PDF)
 
 
+### <a id="data-structures-and-algorithms"></a>Data Structures and Algorithms
+
+* [Data structures and algorithms study cheatsheets for coding interviews](https://www.techinterviewhandbook.org/algorithms/study-cheatsheet/)
+* [Data Structures and Algorithms Cheat Sheet - Cheatography](https://cheatography.com/burcuco/cheat-sheets/data-structures-and-algorithms/)
+* [Algorithms and Data Structures Cheatsheet](https://algs4.cs.princeton.edu/cheatsheet/)
+* [An Executable Data Structures Cheat Sheet for Interviews](https://algodaily.com/lessons/an-executable-data-structures-cheat-sheet)
+* [Big O Notation Cheat Sheet](https://algodaily.com/lessons/big-o-notation-cheat-sheet)
```
Comment (metabiswadeep) [inline more/free-programming-cheatsheets.md:91]:
```suggestion
* [Algorithms and Data Structures Cheatsheet](https://algs4.cs.princeton.edu/cheatsheet/)
* [An Executable Data Structures Cheat Sheet for Interviews](https://algodaily.com/lessons/an-executable-data-structures-cheat-sheet)
* [Big O Notation Cheat Sheet](https://algodaily.com/lessons/big-o-notation-cheat-sheet)
* [Data Structures and Algorithms Cheat Sheet - Cheatography](https://cheatography.com/burcuco/cheat-sheets/data-structures-and-algorithms/)
* [Data structures and algorithms study cheatsheets for coding interviews](https://www.techinterviewhandbook.org/algorithms/study-cheatsheet/)
```

**Review by metabiswadeep — CHANGES_REQUESTED** [review]

File: more/free-programming-cheatsheets.md (hunk 1 above)
Comment (gupta-soham) [reply more/free-programming-cheatsheets.md]: Thanks a ton for the help @metabiswadeep 😅

**Comment by gupta-soham** [comment]
can the owner or the reviewer pull/merge my PR @eshellman @metabiswadeep.Thanks!

File: more/free-programming-cheatsheets.md [hunk 2]
```diff
@@ -81,6 +82,15 @@
 * [Cheatsheets for Data Scientists](https://www.datacamp.com/community/data-science-cheatsheets) - Datacamp (PDF)
 
 
+### <a id="data-structures-and-algorithms"></a>Data Structures and Algorithms
```
Comment (eshellman) [inline more/free-programming-cheatsheets.md:85]: for future reference, the anchor is not needed, it gets generated by the markdown processor.

**Comment by eshellman** [comment]
Thanks, and welcome to free-programming-books!

**Pull request merged**